packetbeat/procs/procs_windows.go

```diff
@@ -196,28 +196,13 @@ func addressIPv6(s [16]byte) net.IP {
 }
 
 func addressIPv4(value uint32) net.IP {
-	// FIXME: This can be rewritten as net.IP((*[4]byte)(unsafe.Pointer(&be))[:])
-	address := make([]byte, 4)
-	machineEndiannes.PutUint32(address, value)
-	return net.IP(address)
-}
-
-var machineEndiannes = getMachineEndiannes()
-
-func getMachineEndiannes() binary.ByteOrder {
-	var buf [2]byte
-	*(*uint16)(unsafe.Pointer(&buf[0])) = 1
-	if buf[0] == 1 {
-		return binary.LittleEndian
-	}
-	return binary.BigEndian
+	return net.IP((*[4]byte)(unsafe.Pointer(&value))[:])
```

Comment:
On the basis of https://docs.microsoft.com/en-us/windows/win32/api/winsock2/ns-winsock2-in_addr I think this is still unnecessarily complex.

If the IPv4 structs in syscall_windows.go replaced the address field with a `type inAddr [4]byte` the memory mapping would be done in the `Extract` methods and this function would become

```
func addressIPv4(s [4]byte) net.IP {
    return s[:]
}
```
similar to the IPv6 case.